### PR TITLE
[SPARK-26853][SQL] Add example and version for commonly used aggregate function descriptions

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?
This improves the expression description for commonly used aggregate functions such as Max, Min, Count, etc.

## How was this patch tested?
Verified the function description manually from the shell.

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you, @dilipbiswal . However, please check the output manually on `spark-sql`. There are many typo bugs related to `_FUNC`. Actually, all newly added description are wrong now. We need to use `_FUNC_` to show the correct function name. It's a place holder.

**Comment by dilipbiswal** [comment]
@dongjoon-hyun Thanks a lot. Very sorry to have missed it. I will fix.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala [hunk 1]
```diff
@@ -24,7 +24,15 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
...
-  usage = "_FUNC_(expr) - Returns the mean calculated from values of a group.")
+  usage = "_FUNC_(expr) - Returns the mean calculated from values of a group.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(col) FROM VALUES (1), (2), (3) AS tab(col);
+       2.0
+      > SELECT _FUNC_(col) FROM VALUES (1), (2), (NULL) AS tab(col);
+       1.5
+  """,
+  since = "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala]: @dilipbiswal . This is `1.0.0`. Please see [SPARK-1251 ](https://issues.apache.org/jira/browse/SPARK-1251).
```scala
scala> sc.version
res15: String = 1.0.0

scala> sqlContext.sql("SELECT avg(age) FROM people WHERE age >= 13 AND age <= 19").collect
res14: Array[org.apache.spark.sql.Row] = Array([19.0])
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala [hunk 2]
```diff
@@ -29,7 +29,17 @@ import org.apache.spark.sql.types._
...
+  examples = """
+    Examples:
+      > SELECT _FUNC_(*) FROM VALUES (NULL), (5), (5), (20) AS tab(col);
+       4
+      > SELECT _FUNC_(col) FROM VALUES (NULL), (5), (5), (20) AS tab(col);
+       3
+      > SELECT _FUNC_(DISTINCT col) FROM VALUES (NULL), (5), (5), (10) AS tab(col);
+       2
+  """,
+  since = "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Max.scala [hunk 3]
```diff
@@ -24,7 +24,13 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
...
 
 @ExpressionDescription(
-  usage = "_FUNC_(expr) - Returns the maximum value of `expr`.")
+  usage = "_FUNC_(expr) - Returns the maximum value of `expr`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(col) FROM VALUES (10), (50), (20) AS tab(col);
+       50
+  """,
+  since = "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Max.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Min.scala [hunk 4]
```diff
@@ -24,7 +24,13 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
...
 
 @ExpressionDescription(
-  usage = "_FUNC_(expr) - Returns the minimum value of `expr`.")
+  usage = "_FUNC_(expr) - Returns the minimum value of `expr`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(col) FROM VALUES (10), (-1), (20) AS tab(col);
+       -1
+  """,
+  since = "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Min.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala [hunk 5]
```diff
@@ -24,7 +24,17 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
...
+  examples = """
+    Examples:
+      > SELECT _FUNC_(col) FROM VALUES (5), (10), (15) AS tab(col);
+       30
+      > SELECT _FUNC_(col) FROM VALUES (NULL), (10), (15) AS tab(col);
+       25
+      > SELECT _FUNC_(col) FROM VALUES (NULL), (NULL) AS tab(col);
+       NULL
+  """,
+  since = "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala (hunk 1 above)
Comment (dilipbiswal) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala]: @dongjoon-hyun Thanks.. will change.

Comment (dongjoon-hyun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala]: Thanks!

**Comment by dilipbiswal** [comment]
retest this please

**Comment by dilipbiswal** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]
Thanks for doing this.

**Comment by dongjoon-hyun** [comment]
@dilipbiswal . As we know, here is [a list of registered aggregation functions](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L280-L312).

Since this PR touchs `approx_percentile`, can we update `approx_count_distinct`, too? In general, the followings are commonly used, too.

- `collect_list`, `collect_set`
- `std (=stddev_samp)`, `stddev_pop`
- `variance (=var_samp)`, `var_pop`
- `covar_samp`, `covar_pop`
- `percentile`

I'm wondering if you can review the list and add examples for all aggregation functions for Apache Spark 3.0.0. How do you think about that?

**Comment by dongjoon-hyun** [comment]
Also, cc @gatorsmile .

**Comment by gatorsmile** [comment]
Please double check the others and complete them, if missing. Thanks!

**Comment by dilipbiswal** [comment]
@dongjoon-hyun Thanks.. Actually, i am aware of missing examples for other functions. It will take me some time to learn the usage of other aggregate functions as i have not used them. Is it possible for us to do this incrementally ?

**Comment by dongjoon-hyun** [comment]
Oh, is that a problem? Then, please use the following examples.

* `collect_list`, `collect_set`
```sql
spark-sql> SELECT collect_set(col) FROM VALUES (1), (2), (1) AS tab(col);
[1,2]
spark-sql> SELECT collect_list(col) FROM VALUES (1), (2), (1) AS tab(col);
[1,2,1]
```

* `std (=stddev_samp)`, `stddev_pop`
```sql
spark-sql> SELECT stddev_samp(col) FROM VALUES (1), (2), (3) AS tab(col);
1.0

spark-sql> SELECT stddev_pop(col) FROM VALUES (1), (2), (3) AS tab(col);
0.816496580927726
```

* `variance (=var_samp)`, `var_pop`
```sql
spark-sql> SELECT var_samp(col) FROM VALUES (1), (2), (3) AS tab(col);
1.0

spark-sql> SELECT var_pop(col) FROM VALUES (1), (2), (3) AS tab(col);
0.6666666666666666
```

* `covar_samp`, `covar_pop`
```sql
spark-sql> SELECT covar_samp(c1,c2) FROM VALUES (1,1), (2,2), (3,3) AS tab(c1,c2);
1.0

spark-sql> SELECT covar_pop(c1,c2) FROM VALUES (1,1), (2,2), (3,3) AS tab(c1,c2);
0.6666666666666666
```

* `percentile`
```sql
spark-sql> SELECT percentile(col, 0.3) FROM VALUES (0), (10) AS tab(col);
3.0

spark-sql> SELECT percentile(col, array(0.25,0.75)) FROM VALUES (0), (10) AS tab(col);
[2.5,7.5]
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please use the examples I gave or in [windows.sql](https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/window.sql#L58-L87).

**Comment by dilipbiswal** [comment]
@dongjoon-hyun So nice of you :-). I will use the examples from you.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @dilipbiswal !

**Comment by dongjoon-hyun** [comment]
Thank you for updating, @dilipbiswal !

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @dilipbiswal , @HyukjinKwon , @gatorsmile !

**Comment by dilipbiswal** [comment]
@dongjoon-hyun @gatorsmile @HyukjinKwon Thank you very much.

**Pull request closed without merging**